Utilities/ocvalidate/ValidatePlatformInfo.c

```diff
@@ -0,0 +1,85 @@
+/** @file
+  Copyright (C) 2018, vit9696. All rights reserved.
+  Copyright (C) 2020, PMheart. All rights reserved.
+
+  All rights reserved.
+
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+**/
+
+#include "ocvalidate.h"
+#include "OcValidateLib.h"
+
+#include <Library/OcMacInfoLib.h>
+
+//
+// NOTE: Only PlatformInfo->Generic is checked here. The rest is ignored.
+//
+UINT32
+CheckPlatformInfo (
+  IN  OC_GLOBAL_CONFIG  *Config
+  )
+{
+  UINT32              ErrorCount;
+  EFI_STATUS          Status;
+  OC_PLATFORM_CONFIG  *UserPlatformInfo;
+  BOOLEAN             IsAutomaticEnabled;
+  CONST CHAR8         *UpdateSMBIOSMode;
+  CONST CHAR8         *SystemProductName;
+  CONST CHAR8         *SystemMemoryStatus;
+  CONST CHAR8         *AsciiSystemUUID;
+  GUID                Guid;
+
+  DEBUG ((DEBUG_VERBOSE, "config loaded into PlatformInfo checker!\n"));
+
+  ErrorCount         = 0;
+  UserPlatformInfo   = &Config->PlatformInfo;
+  IsAutomaticEnabled = UserPlatformInfo->Automatic;
+  UpdateSMBIOSMode   = OC_BLOB_GET (&UserPlatformInfo->UpdateSmbiosMode);
+  SystemProductName  = OC_BLOB_GET (&UserPlatformInfo->Generic.SystemProductName);
+  SystemMemoryStatus = OC_BLOB_GET (&UserPlatformInfo->Generic.SystemMemoryStatus);
+  AsciiSystemUUID    = OC_BLOB_GET (&UserPlatformInfo->Generic.SystemUuid);
+
+  if (!IsAutomaticEnabled) {
+    DEBUG ((DEBUG_WARN, "PlatformInfo->Automatic is not enabled!\n"));
```

Comment:
This is not a mistake. Please silently quit in this case.

Reply:
https://github.com/PMheart/OpenCorePkg/commit/80448d11c0e6ff9860a9301c64435307755be8cf